pkg/controller/elasticsearch/driver/nodes.go

```diff
@@ -66,6 +70,7 @@ func (d *defaultDriver) reconcileNodeSpecs(
 	}
 	actualStatefulSets, err = HandleUpscaleAndSpecChanges(upscaleCtx, actualStatefulSets, expectedResources)
 	if err != nil {
+		reconcileState.AddEvent(corev1.EventTypeWarning, events.EventReconciliationError, fmt.Sprintf("Failed to apply spec change: %v", err))
```

Comment:
This will also catch errors such as:
- cannot update the ES config
- cannot request ELasticsearch to grab the number of master nodes
- cannot reconcile the sset service

I'm wondering whether we should explicitly compare actual vs. expected VolumeClaimTemplates in the code and emit event + log at that time only. Maybe in a longer-term, fine with keeping this as a short-term solution.